std/functional.d

```diff
@@ -1847,3 +1847,168 @@ if (isCallable!(F))
         static assert(! is(typeof(dg_xtrnC) == typeof(dg_xtrnD)));
     }
 }
+
+// Converts an unsigned integer to a compile-time string constant.
+private enum toCtString(ulong n) = n.stringof[0 .. $ - "LU".length];
+
+// Check that .stringof does what we expect, since it's not guaranteed by the
+// language spec.
+@safe unittest
+{
+    assert(toCtString!0 == "0");
+    assert(toCtString!123456 == "123456");
+}
```

Comment:
Why not static asserts instead?

Reply:
This is copy+pasted from `std.sumtype`, and it uses runtime `assert` there too.

Reply:
Can't this be in a shared module? I don't see the reason for a runtime assert.

Reply:
Once this is merged I plan to submit a separate refactoring PR addressing these issues. Since this PR introduces a new feature, I want to keep refactoring out of it.